### PR TITLE
ligo: 0.60.0 -> 0.64.2

### PR DESCRIPTION
**Pull request by Marigold-infrabot**
###### Description of changes 

Update ligo version to latest one 

###### Things done 

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 228539` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ocamlPackages.ringo-lwt</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>ligo</li>
    <li>ocamlPackages.aches</li>
    <li>ocamlPackages.aches-lwt</li>
    <li>ocamlPackages.asetmap</li>
    <li>ocamlPackages.bls12-381-hash</li>
    <li>ocamlPackages.data-encoding</li>
    <li>ocamlPackages.hacl-star</li>
    <li>ocamlPackages.hacl-star-raw</li>
    <li>ocamlPackages.linol</li>
    <li>ocamlPackages.linol-lwt</li>
    <li>ocamlPackages.polynomial</li>
    <li>ocamlPackages.prometheus</li>
    <li>ocamlPackages.ringo</li>
    <li>ocamlPackages.tezos-bls12-381-polynomial</li>
    <li>ocamlPackages.tezos-plompiler</li>
    <li>ocamlPackages.tezos-plonk</li>
    <li>tezos-rust-libs</li>
  </ul>
</details>

**Comment by ulrikstrid** [comment]
`ringo-lwt` is not a thing anymore, removed it

**Comment by ulrikstrid** [comment]
Result of `nixpkgs-review pr 228539` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>ligo</li>
    <li>ocamlPackages.aches</li>
    <li>ocamlPackages.aches-lwt</li>
    <li>ocamlPackages.asetmap</li>
    <li>ocamlPackages.bls12-381-hash</li>
    <li>ocamlPackages.data-encoding</li>
    <li>ocamlPackages.hacl-star</li>
    <li>ocamlPackages.hacl-star-raw</li>
    <li>ocamlPackages.linol</li>
    <li>ocamlPackages.linol-lwt</li>
    <li>ocamlPackages.polynomial</li>
    <li>ocamlPackages.prometheus</li>
    <li>ocamlPackages.ringo</li>
    <li>ocamlPackages.tezos-bls12-381-polynomial</li>
    <li>ocamlPackages.tezos-plompiler</li>
    <li>ocamlPackages.tezos-plonk</li>
    <li>tezos-rust-libs</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 228539` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ligo</li>
    <li>tezos-rust-libs</li>
  </ul>
</details>

File: pkgs/development/libraries/tezos-rust-libs/default.nix [hunk 1]
```diff
@@ -1,37 +1,48 @@
...
+    cargo build \
+      --target-dir target-wasmer \
+      --package wasmer-c-api \
+      --no-default-features \
+      --features singlepass,cranelift,wat,middlewares,universal \
+      --release
   '';
 
-  doCheck = true;
+  installPhase = ''
```
Comment (superherointj) [inline pkgs/development/libraries/tezos-rust-libs/default.nix:36]:
```suggestion
  installPhase = ''
```
runHook missing

File: pkgs/development/ocaml-modules/hacl-star/raw.nix [hunk 2]
```diff
@@ -21,18 +29,26 @@ stdenv.mkDerivation rec {
     patchShebangs ./
   '';
 
-  preInstall = ''
-    mkdir -p $OCAMLFIND_DESTDIR/stublibs
+  buildPhase = ''
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/hacl-star/raw.nix:32]: runHook missing

File: pkgs/development/ocaml-modules/hacl-star/raw.nix [hunk 3]
```diff
@@ -21,18 +29,26 @@ stdenv.mkDerivation rec {
...
+    make -C hacl-star-raw build-c
+    make -C hacl-star-raw build-bindings
   '';
 
-  installTargets = "install-hacl-star-raw";
+  installPhase = ''
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/hacl-star/raw.nix:37]: runHook missing

**Pull request merged**